src/Microsoft.Diagnostics.Runtime/src/Common/ModuleInfo.cs

```diff
@@ -11,11 +11,14 @@ namespace Microsoft.Diagnostics.Runtime
     /// <summary>
     /// Provides information about loaded modules in a <see cref="DataTarget"/>.
     /// </summary>
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
     public class ModuleInfo
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
     {
         private readonly IDataReader _dataReader;
         private bool? _isManaged;
         private VersionInfo? _version;
+        internal bool _isVirtual;
```

Comment:
If this is writable it should be a public property.  By making it an internal variable then someone couldn't (reasonably) write their own DataReader similar to LinuxLiveDataReader.

As an alternative, it looks like the only place we set this value is LinuxLiveDataReader?  Without having deeply read through this change, maybe it's possible to recreate ModuleInfo in LinuxLiveDataReader instead of updating the value?  I'm not sure if that's feasible, just brainstorming.

Reply:
> it looks like the only place we set this value is LinuxLiveDataReader?

In Linux part, yes. If we go this route on Windows then `LiveDataReader` and `DbgEngDataReader` may have similar behaviors.

> maybe it's possible to recreate ModuleInfo in LinuxLiveDataReader instead of updating the value?

I assume recording these values in a collection and creating `ModuleInfo`s in the end could have lower overheads than recreating.

Reply:
> it looks like the only place we set this value is LinuxLiveDataReader?

In Linux part, yes. If we go this route on Windows then `LiveDataReader` and `DbgEngDataReader` may have similar behaviors.

> maybe it's possible to recreate ModuleInfo in LinuxLiveDataReader instead of updating the value?

I assume recording these values in a collection and creating `ModuleInfo`s in the end could have lower overheads than recreating.

Reply:
> I assume recording these values in a collection and creating `ModuleInfo`s in the end could have lower overheads than recreating.

That seems like the best approach.

Reply:
Then we'll arrive at a design with a private read-only field. In both cases, the users only see a parameter. However, by making the field internally writable, we have additional benefits of reducing recreating or other overheads in our internal implementation.

Reply:
Can you comment on this? The internal field looks similar to that internal method of `ElfLoadedImage` (although it's not a public type). By making it internal we do make sure it's initialized only so `GetPEImage` is guaranteed to return the correct result. The users could achieve the same by recreating or other one-time creation approaches.

https://github.com/microsoft/clrmd/blob/daafcb3fa7426526237a5fecccee042a0d243996/src/Microsoft.Diagnostics.Runtime/src/Linux/ElfLoadedImage.cs#L50

I have no objection to making it public, just want to think more about it.